### ISSUE TITLE
`NamedTuple` is not satisfying `Hashable` constraint

### ISSUE THREAD

**Issue by aslpavel** [microsoft/pyright#3141]
Any instance of `NamedTuple` should satisfy generic bound `Hashable`

**To Reproduce**
```python
# pyright: strict

from typing import NamedTuple, TypeVar
from collections.abc import Hashable


class A(NamedTuple):
    attr: int


T = TypeVar("T", bound=Hashable)


def gen(a: T) -> T:
    print(a.__hash__())
    return a


value = A(127)
gen(value)
assert isinstance(value, Hashable)
```
```
$ pyright main.py
hashable-bug/main.py
  hashable-bug/main.py:20:5 - error: Argument of type "A" cannot be assigned to parameter "a" of type "T@gen" in function "gen"
    Type "A" cannot be assigned to type "Hashable" (reportGeneralTypeIssues)
$ mypy --strict main.py 
Success: no issues found in 1 source file
```

**Expected behavior**
Should type check without errors

**VS Code extension or command-line**
Reproducible with
**Pylance:** v2022.2.4
**Pyright:** 1.1.223
**Python:** 3.10.2

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.227, which I just published. It will also be included in the next release of pylance.
